src/vs/workbench/contrib/chat/browser/actions/chatActions.ts

```diff
@@ -439,6 +439,69 @@ abstract class ModeOpenChatGlobalAction extends OpenChatGlobalAction {
 	}
 }
 
+class SwitchCustomChatAgentAction extends OpenChatGlobalAction {
+	static readonly ID = 'workbench.action.chat.switchCustomChatAgent';
+
+	constructor() {
+		super({
+			id: SwitchCustomChatAgentAction.ID,
+			title: localize2('switchCustomChatAgent', "Switch to Custom Chat Agent"),
+			keybinding: undefined
+		});
+	}
+
+	override async run(accessor: ServicesAccessor, opts?: string | IChatViewOpenOptions): Promise<IChatAgentResult & { type?: 'confirmation' } | undefined> {
+		opts = typeof opts === 'string' ? { query: opts } : opts;
+
+		const chatModeService = accessor.get(IChatModeService);
+		const quickInputService = accessor.get(IQuickInputService);
+		const commandService = accessor.get(ICommandService);
+
+		const modes = chatModeService.getModes().custom;
+		if (modes.length === 0) {
+			return undefined;
+		}
+
+		let selectedModeName: string | undefined;
+		const agentName = opts?.mode;
+		if (agentName) {
+			// If agent name is provided as argument, use it directly
+			const selectedMode = modes.find(m => m.name === agentName);
```

Comment:
The ToggleChatModeAction handles this case of switching to a mode by id, so users can already set up their keybindings that way. The reason I wanted separate actions vs one action with args is that this way is not discoverable at all, and we have no UX for it in the keyboard shortcuts editor.